File: common/src/main/java/com/taskadapter/connector/common/RelationSaver.java
package com.taskadapter.connector.common;

import com.taskadapter.connector.definition.exceptions.ConnectorException;
import com.taskadapter.model.GRelation;

import java.util.List;

public interface RelationSaver {
    /**
     * Save task relations.
     *
     * @param relations relations to save.
     * @throws ConnectorException if connector fails to save a relation.
     */
    void saveRelations(List<GRelation> relations) throws ConnectorException;
}

File: common/src/main/scala/com/taskadapter/connector/common/RelationSaver.scala
package com.taskadapter.connector.common

import java.util

import com.taskadapter.connector.definition.exceptions.ConnectorException
import com.taskadapter.model.GRelation

trait RelationSaver {
  /**
    * Save task relations.
    *
    * @param relations relations to save.
    * @throws ConnectorException if connector fails to save a relation.
    */
  @throws[ConnectorException]
  def saveRelations(relations: util.List[GRelation]): Unit
}
